src/sys/resource.rs

```diff
@@ -0,0 +1,48 @@
+use std::mem;
+
+use libc::{self, c_int};
+
+use {Errno, Result};
+
+#[repr(i32)]
+pub enum Resource {
```

Comment:
If we add the `get` and `set` methods to the type, the name `ResourceLimit` does make more sense. However, if we don't do that, the variants of the enum actually represent the Resource, whose limit is set/get by the methods `setrlimit`, `getrlimit`.

The more I think about it, the more I like @posborne's proposal.